modules/check_in/lib/check_in/utils/logger.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module CheckIn
+  module Utils
+    class Logger
+      API_STATUS_REGEXP = /\W+status\W+([^"|^,|^\\]*)/.freeze
+
+      attr_reader :ctrl
+
+      def self.build(controller)
+        new(controller)
+      end
+
+      def initialize(controller)
+        @ctrl = controller
+      end
+
+      def before
+        common.merge({ filter: :before_action })
+      end
+
+      def after
+        common.merge({ api_status: api_status, filter: :after_action })
+      end
+
+      def common
+        {
+          workflow: workflow,
+          uuid: uuid,
+          controller: ctrl_name,
+          action: ctrl_action
+        }
+      end
+
+      def ctrl_name
+        ctrl.controller_name
+      end
+
+      def ctrl_action
+        ctrl.action_name
+      end
+
+      def api_status
+        body = ctrl.response.body
+
+        return 'none' if body.blank?
+
+        body.match(API_STATUS_REGEXP) { |g| g[1] if g }
```

Comment:
just noting our discussion for posterity: we discussed the concern about parsing response body on every action call, and that it would be mitigated once we update the controllers to return an appropriate HTTP response code for each request (which will require working with FE team and Product team on defining what response code should be returned based on downstream responses).

Reply:
Thanks!